### PR TITLE
Added --block option for blocking job submission

### PR DESCRIPTION
**Pull request by Gab-D-G**
…r PBS and '--wait' for SLURM.

### CODE REVIEW

File: qbatch/qbatch.py [hunk 1]
```diff
@@ -369,6 +369,15 @@ def qbatchDriver(**kwargs):
     system = kwargs.get('system')
     env_mode = kwargs.get('env')
     shell = kwargs.get('shell')
+    block = kwargs.get('block')
+
+    if block:
+        if system == 'sge':
+            options.append('sync -y')
```
Comment (gdevenyi) [inline qbatch/qbatch.py:376]: This should be `-sync y`

File: qbatch/qbatch.py [hunk 2]
```diff
@@ -750,6 +759,10 @@ def qbatchParser(args=None):
         "--shell", default=SHELL,
         help="""Shell to use for spawning jobs
         and launching single commands""")
+    group.add_argument(
+        "--block", action="store_true",
+        help="""For SGE, PBS and SLURM, blocks execution until jobs are finished.
+        'sync -y' is called for SGE, '-Wblock=true' for PBS and '--wait' for SLURM.""")
```
Comment (gdevenyi) [inline qbatch/qbatch.py:765]: We don't need these details here.

File: qbatch/qbatch.py [hunk 3]
```diff
@@ -369,6 +369,15 @@ def qbatchDriver(**kwargs):
...
     shell = kwargs.get('shell')
+    block = kwargs.get('block')
+
+    if block:
+        if system == 'sge':
+            options.append('sync -y')
+        elif system == 'pbs':
+            options.append('-Wblock=true')
+        elif system == 'slurm':
+            options.append('--wait')
```
Comment (gdevenyi) [inline qbatch/qbatch.py:380]: These appends need to have a space at the front of them so they don't concat against the other options and break everything.

**Review by gdevenyi — CHANGES_REQUESTED** [review]
This is close to what I was expecting, but not quite. This should be its own option within the run templates.

See for example, `o_walltime`

in the templates:
https://github.com/pipitone/qbatch/blob/master/qbatch/qbatch.py#L93
https://github.com/pipitone/qbatch/blob/master/qbatch/qbatch.py#L115
https://github.com/pipitone/qbatch/blob/master/qbatch/qbatch.py#L136

And then filled in later in the logic:
https://github.com/pipitone/qbatch/blob/master/qbatch/qbatch.py#L448
https://github.com/pipitone/qbatch/blob/master/qbatch/qbatch.py#L469
https://github.com/pipitone/qbatch/blob/master/qbatch/qbatch.py#L483-L489

File: qbatch/qbatch.py (hunk 1 above)
Comment (Gab-D-G) [reply qbatch/qbatch.py:376]: what do you mean? it should be between ``` ?

Comment (gdevenyi) [reply qbatch/qbatch.py:376]: The option is `-sync y`.

```
     -sync y[es]|n[o]
          Available for qsub.

          -sync y causes qsub to wait for  the  job  to  complete
          before  exiting.   If  the  job completes successfully,
          qsub's exit code will be that of the completed job.  If
          the job fails to complete successfully, qsub will print
          out a error message indicating why the job  failed  and
          will  have  an exit code of 1.  If qsub is interrupted,
          e.g. with CTRL-C, before the  job  completes,  the  job
          will be canceled.
          With the -sync n option, qsub will exit  with  an  exit
          code of 0 as soon as the job is submitted successfully.
          -sync n is default for qsub.
          If -sync y is used in conjunction  with  -now  y,  qsub
          will  behave as though only -now y were given until the
          job has been successfully scheduled, after  which  time
          qsub will behave as though only -sync y were given.
          If -sync y is used in conjunction  with  -t  n[-m[:i]],
          qsub  will  wait  for  all  the job's tasks to complete
          before exiting.  If all the job's tasks  complete  suc-
          cessfully,  qsub's  exit code will be that of the first
          completed job tasks with a non-zero exit code, or 0  if
          all job tasks exited with an exit code of 0.  If any of
          the job's tasks fail  to  complete  successfully,  qsub
          will  print out an error message indicating why the job
          task(s) failed and will have an exit  code  of  1.   If
          qsub  is  interrupted, e.g. with CTRL-C, before the job
```

Comment (Gab-D-G) [reply qbatch/qbatch.py]: ah my bad, I'll fix this

File: qbatch/qbatch.py (hunk 3 above)
Comment (Gab-D-G) [reply qbatch/qbatch.py]: Fixed

File: qbatch/qbatch.py (hunk 2 above)
Comment (Gab-D-G) [reply qbatch/qbatch.py]: Let me know if the update is satisfying there.

Comment (gdevenyi) [reply qbatch/qbatch.py]: The `o_block` definitions should be the inline assignment type like
https://github.com/pipitone/qbatch/blob/master/qbatch/qbatch.py#L448

Currently, if block is undefined, o_block will not exist, and the template filling will fail.

Comment (Gab-D-G) [reply qbatch/qbatch.py]: This should be addressed with the latest commit.

**Review by gdevenyi — APPROVED** [review]

**Comment by gdevenyi** [comment]
This is good to go, but it seems we've lost our Travis CI instance. I'm waiting on @pipitone to re-enable so we can make sure the tests pass.

**Comment by gdevenyi** [comment]
hurrah! thanks

**Pull request merged**